src/feature/UserList/AddUsers/AddUsers.tsx

```diff
@@ -0,0 +1,178 @@
+
+  const createUser = () => {
+    apiClient()
+      .post('users', formValues)
+      .then(() => {
+        dispatch(loadUsersAsync())
+      })
+  }
+
+  const onFinish = (values: FormData) => {
+    setFormValues(initialData(values))
+  }
+
+  useEffect(() => {
+    if (isLoading && isModalOpen) {
+      setIsDisabled(true)
+    } else if (!isLoading && isModalOpen) {
+      setModalState(false)
+      setIsDisabled(false)
+    }
+  }, [isLoading])
+
+  useEffect(() => {
+    if (formValues) {
+      createUser()
+    }
+  }, [formValues])
+
+  return (
+    <div>
+      <Button onClick={toggleModal}>Create user</Button>
+      <Modal title={'Create user'} isOpen={isModalOpen} onClose={toggleModal}>
+        <Form name="complex-form" onFinish={onFinish}>
+          <Form.Item
+            label="First name"
+            name="first_name"
+            rules={[
+              {
+                required: true,
+                message: 'First name is required',
+              },
+            ]}
+          >
+            <Input
+              disabled={isDisabled}
+              autoComplete="new-password"
+              placeholder="Please input first name"
+            />
+          </Form.Item>
+          <Form.Item
+            label="Last name"
+            name="last_name"
+            rules={[
+              {
+                required: true,
+                message: 'Last name is required',
+              },
+            ]}
+          >
+            <Input
+              disabled={isDisabled}
+              autoComplete="new-password"
+              placeholder="Please input last name"
+            />
+          </Form.Item>
+          <Form.Item
+            name="user_name"
+            label="User name"
+            rules={[
+              {
+                required: true,
+                message: 'User name is required',
+              },
+            ]}
+          >
+            <Input
+              disabled={isDisabled}
+              autoComplete="new-password"
+              placeholder="Please input user name"
+            />
+          </Form.Item>
+          <Form.Item
+            name="email"
+            label="E-mail"
+            rules={[
+              {
+                type: 'email',
+                message: 'The input is not valid E-mail!',
+              },
+              {
+                required: true,
+                message: 'Please input your e-mail!',
+              },
+            ]}
+          >
+            <Input
+              placeholder="Please input email"
+              autoComplete="new-password"
+              disabled={isDisabled}
+            />
+          </Form.Item>
+          <Form.Item
+            label="Role"
+            name="role"
+            rules={[
+              {
+                required: true,
+                message: 'Role is required',
+              },
+            ]}
+          >
+            <Select disabled={isDisabled} placeholder="Select role">
```

Comment:
you can use "options" parameter with type "{label, value}[]" instead on passing jsx